src/view.rs

```diff
@@ -719,25 +719,43 @@ pub fn view_grid_grammar(m: &Model, coord: &Coordinate, sub_coords: Vec<Coordina
 
 pub fn view_context_menu(m: &Model) -> Html {
     let default_options = vec![
-        ("Save", m.link.callback(|_| Action::SaveSession())),
-        ("Zoom In (+)", m.link.callback(|_| Action::ZoomIn)),
-        ("Zoom Reset", m.link.callback(|_| Action::ZoomReset)),
-        ("Zoom Out (-)", m.link.callback(|_| Action::ZoomOut)),
-        ("Reset", m.link.callback(|_| Action::Recreate)),
-        ("Insert Row", m.link.callback(|_| Action::InsertRow)),
-        ("Insert Col", m.link.callback(|_| Action::InsertCol)),
-        ("Merge", m.link.callback(|_| Action::MergeCells())),
-        ("Delete Row", m.link.callback(|_| Action::DeleteRow)),
-        ("Delete Col", m.link.callback(|_| Action::DeleteCol)),
+        ("Save", m.link.callback(|_| Action::SaveSession()), true),
+        ("Zoom In (+)", m.link.callback(|_| Action::ZoomIn), true),
+        ("Zoom Reset", m.link.callback(|_| Action::ZoomReset), true),
+        ("Zoom Out (-)", m.link.callback(|_| Action::ZoomOut), true),
+        ("Reset", m.link.callback(|_| Action::Recreate), true),
+        ("Insert Row", m.link.callback(|_| Action::InsertRow), true),
+        ("Insert Col", m.link.callback(|_| Action::InsertCol), true),
+        ("Merge", m.link.callback(|_| Action::MergeCells()), false),
+        ("Delete Row", m.link.callback(|_| Action::DeleteRow), true),
+        ("Delete Col", m.link.callback(|_| Action::DeleteCol), true),
     ];
     let option_nodes = {
         let mut v = VList::new();
-        for (option_name, option_action) in default_options {
-            v.add_child(html! {
-                <li class="context-menu-option" onclick=option_action>
-                    { option_name }
-                </li>
-            });
+        for (option_name, option_action, option_param) in default_options {
+            let mut param = true;
+            
+            //Switch exceptions
```

Comment:
could you add a longer explanation of what you're doing here? i don't quite understand by just looking at it.

Reply:
it seems like these refer to options that should only show based on conditions like the selection. try to make that as obvious as possible in the code by using more obvious variable names, or adding more comments